mock/src/lib.rs

```diff
@@ -1,5 +1,33 @@
-//! Mock request dispatcher and credentials for unit testing services
-
+//! Mock request dispatcher and credentials for unit testing rusoto AWS service clients
+//!
+//! All rusoto generated service clients come with a constuctor function named `new_with` which
+//! accepts three arguments
+//!
+//! * A `rusoto_core::DispatchSignedRequest` implementation
+//! * A `rusoto_core::credential::ProvideAwsCredentials` implementation
+//! * A `rusoto_core::Region`
+//!
+//! This crate provides mock implementations to satisfy the first two.
+//!
+//! # Example
+//!
+//! ```rust
+//! extern crate rusoto_mock;
+//! // extern crate rusoto_s3;
```

Comment:
Ah, are these lines commented out so we don't have circular dependencies?

Reply:
Precisely. The s3 crate depends on the already and I wanted a way to independently document how you can use wire mock crate into any service test. I'll make an extra note.